### ISSUE TITLE
Download Zoom

### ISSUE THREAD

**Issue by 89z** [fluminurs/fluminurs#667]
Can this download from Zoom? URL like this:

https://us06web.zoom.us/rec/share/5Qb3qHy_bi9WcGazy4W5j80PrQkOpbNIrgW-6vrRkRb5Q2xyxw7RWTJNG2KcO4J4.K-16KTWChtQT-R2h

**Comment by btzy** [comment]
This tool does not bypass Zoom authentication.  You still need to know the password.

**Comment by 89z** [comment]
Assuming I have the passcode (I do), is it possible? I am not trying to do any bypass, I would just like to download content I have access to.

**Comment by btzy** [comment]
Yes, there is code to download Zoom videos assuming you know the passcode, but it is quite coupled to the rest of the code that accesses our school's content management system.  So you won't be able to use this tool directly - you will need to extract the relevant code and build it yourself.

Take a look at this code, which gets the download URL of the video, given the share URL (the one you have in your original comment) and the passcode:
https://github.com/fluminurs/fluminurs/blob/63a1b0745ed1f973bf0bbe1c4158a97c7d142112/src/conferencing.rs#L226-L323

If you need to log in to Zoom via SSO prior to watching the video, then look at this code:
https://github.com/fluminurs/fluminurs/blob/63a1b0745ed1f973bf0bbe1c4158a97c7d142112/src/lib.rs#L364-L369

**Comment by francisyzy** [comment]
There is a [Firefox Extension](https://addons.mozilla.org/en-US/firefox/addon/zed-zoom-easy-downloader/) to download from zoom's page itself. Not a friendly workaround but yea
